p2panda-rs/src/storage_provider/traits/document_store.rs

```diff
@@ -2,55 +2,40 @@
 
 use async_trait::async_trait;
 
-use crate::document::{Document, DocumentId, DocumentView, DocumentViewId};
+use crate::document::traits::AsDocument;
+use crate::document::{DocumentId, DocumentViewId};
 use crate::schema::SchemaId;
 use crate::storage_provider::error::DocumentStorageError;
 
-/// Storage traits for documents and document views.
+use super::OperationStore;
+
+/// Interface for querying `Documents` from the store.
```

Comment:
I'm wondering if removing `insert_docment_view` from this store trait was actually a mistake... It is very neat to only have the API outlining the query methods for documents, but now already storage provider implementations will diverge in their expected behaviour. For example, the SQL store in aquadoggo requires an extra materialisation step after insertion of operations, but `MemoryStore` doesn't. 

Realise this is a "future" problem, but makes me wonder if this design decision is right.